spring-integration-azure/src/main/java/com/microsoft/azure/spring/integration/storage/queue/StorageQueueTemplate.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See LICENSE in the project root for
+ * license information.
+ */
+
+
+package com.microsoft.azure.spring.integration.storage.queue;
+
+import com.microsoft.azure.spring.integration.storage.queue.factory.StorageQueueFactory;
+import com.microsoft.azure.storage.StorageException;
+import com.microsoft.azure.storage.queue.CloudQueue;
+import com.microsoft.azure.storage.queue.CloudQueueMessage;
+import org.springframework.util.Assert;
+
+
+public class StorageQueueTemplate implements StorageQueueOperation {
+    private final StorageQueueFactory storageQueueFactory;
+
+    StorageQueueTemplate(StorageQueueFactory storageQueueFactory) {
+        this.storageQueueFactory = storageQueueFactory;
+    }
+
+    private CloudQueue getOrCreateQueue(String destination) {
+        Assert.hasText(destination, "destination can't be null or empty");
+        return storageQueueFactory.getQueueCreator().apply(destination);
+    }
+
+    @Override
+    public boolean add(String destination, CloudQueueMessage cloudQueueMessage) {
+        CloudQueue cloudQueue = getOrCreateQueue(destination);
+        try {
+            cloudQueue.addMessage(cloudQueueMessage);
+        } catch (StorageException e) {
+            throw new StorageQueueRuntimeException("Failed to add message to cloud queue", e);
+        }
+        return true;
```

Comment:
If this method always return true, why we need boolean return value?

Reply:
I wrote this method refer to java.util.Queue. It return true or throw exception. We need boolean return value to decide what to do next.